### PR TITLE
Added new Numeric Transform - LogFunction

### PR DESCRIPTION
**Pull request by GaryWSmith**
This Transform applies the log function to an input i.e. output = (multiplier * LN(input) + offset)
It has been copied directly from a similar Transform so hopefully it meets all of the guidelines.
It is useful as a simple method of characterising NTC thermistors although being generic it has many potential uses.

### CODE REVIEW

File: src/transforms/logfunction.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+LogFunction::LogFunction(float constant_1, float constant_2,
+                             String config_path)
+    : NumericTransform(config_path),
+      multiplier{multiplier},
+      offset{offset}{
+  className = "LogFunction";
+  load_configuration();
+}
+
```
Comment (ba58smith) [inline src/transforms/logfunction.cpp:11]: I think you mean for constant_1 and constant_2 to be multiplier and offset, don't you?

File: src/transforms/logfunction.cpp [hunk 2]
```diff
@@ -23,8 +23,8 @@ void LogFunction::get_configuration(JsonObject& root) {
 static const char SCHEMA[] PROGMEM = R"({
     "type": "object",
     "properties": {
-        "constant_1": { "title": "multiplier", "type": "number", "description": "The (m)ultiplier in the equation f(x) = m*log(x) + o" },
-        "offset": { "title": "offset", "type": "number", "description": "The (o)ffset in the equation f(x) = m*log(x) + 0" },
```
Comment (ba58smith) [inline src/transforms/logfunction.cpp]: I just noticed this for the first time: right above this comment, shouldn't "constant_1" be "multiplier"? Sorry I didn't catch it before.

**Comment by ba58smith** [comment]
I also just noticed that you created this PR on your "master" branch, rather than on a specific branch created for this PR. As noted in the wiki: 

"For your own benefit, it is recommended that you create a new Git branch for each PR. It doesn't necessarily change the end result, but it's very likely that you'll be asked to make some changes in the PR, and if you're working out of your master branch, you might end up in a world of pain. Hence, make a unique branch for each PR. ***If you don't already have a Git process that works perfectly, and that includes making a new branch for each PR, then please follow the steps below exactly.***"

So, please put this PR into a new branch, rather than have it in "master".

**Comment by GaryWSmith** [comment]
I did, thank you Brian.

**Comment by GaryWSmith** [comment]
Re the specific branch - I'll close this and try again.

**Pull request closed without merging**